src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,112 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String CREATE_BOOK_QUERY
+            = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String FIND_BY_ID_QUERY = "SELECT * FROM books WHERE id = ?";
+    private static final String FIND_ALL_QUERY = "SELECT * FROM books";
+    private static final String UPDATE_BOOK_QUERY
+            = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BY_ID_QUERY = "DELETE FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(CREATE_BOOK_QUERY,
+                                Statement.RETURN_GENERATED_KEYS)) {
+
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.executeUpdate();
+
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                book.setId(resultSet.getLong(1));
+            }
+            return book;
+
+        } catch (SQLException e) {
```

Comment:
Use informative exception messages. Instead of 'Failed to insert book into DB', include the book details to make the exception message more informative.